diff --git a/node_modules/cosmos-tx-builder/index.js b/node_modules/cosmos-tx-builder/index.js
index c766efb..a7e7021 100644
--- a/node_modules/cosmos-tx-builder/index.js
+++ b/node_modules/cosmos-tx-builder/index.js
@@ -123,7 +123,7 @@ const sign_transaction = async function(jsonTx, wallet, sequence, account_number

         const signMessage = await create_sign_message(jsonTx, sequence, account_number, chain_id)

-        const signatureBuffer = await sign_with_privkey(signMessage, wallet.privateKey)
+        const signatureBuffer = await sign_with_privkey(signMessage, wallet.privateKey !== undefined ? wallet.privateKey : wallet)

         const pubKeyBuffer = Buffer.from(wallet.publicKey, `hex`)

@@ -198,11 +198,14 @@ const create_sign_message = async function(jsonTx,sequence, account_number, chai
 const sign_with_privkey = async function(signMessage, privateKey){
     let tag = TAG+" | sign_with_privkey | "
     try{
-        console.log(tag,"signMessage: ",typeof(signMessage))
-        console.log(tag,"signMessage: ",signMessage)
-
-        const signHash = Buffer.from(sha256(signMessage).toString(), `hex`)
-        const { signature } = secp256k1.sign(signHash, Buffer.from(privateKey, `hex`))
+        // console.log(tag,"signMessage: ",typeof(signMessage))
+        // console.log(tag,"signMessage: ",signMessage)
+
        const signature = (typeof privateKey.sign === "function" ? privateKey.sign(signMessage) : (() => {
            const signHash = Buffer.from(sha256(signMessage).toString(), `hex`);
            return secp256k1.sign(signHash, Buffer.from(privateKey, `hex`)).signature;
        })());
+        if (typeof signature.then === "function") signature = await signature;

         //log.debug(tag,"signature: ",signature)

